Makefile

```diff
@@ -64,16 +64,6 @@ else
    # export GOOS ?= windows
 endif
 
-ifeq ($(GOOS),linux)
```

Comment:
@mattdelco so that section isn't needed? will it still build native on mac ?

Reply:
The section was to use 'lx/amd65' ( not sure why ), I don't think it has any effect on mac since GOOS is not set explicitly. If it breaks anything we can fix it - we don't have any mac presubmit ( circleci does support mac builds actually, but not sure how they could be turned on )

Reply:
The section was an attempt to keep the directories short (mostly for situations without auto-complete), and for anyone who's been hiding in a cave and didn't know what "darwin" was.  I suppose if I really cared about total length I could've also used "rel" and "dbg" for BUILDTYPE_DIR.  Most cross-compiled builds I've seen used a 2 or 3 level directory structure akin to what I used.  The only time I can recall underscores is in the Windows driver dev kit (and that might've been to avoid a 4th directory for OS version).

Anyway, I'm not wedded to any particular scheme--I just wish it could've been hashed out earlier in my PR so that a PR like this didn't cause a rebuild and stale directories for devs.